__tests__/app.test.js

```diff
@@ -40,65 +40,154 @@ describe("NC News", () => {
+
+          expect(typeof article.created_at).toBe('string');
+          expect(typeof article.article_img_url).toBe('string');
+
+          // Expect the article to have the expected set of properties and no others.
+
+          const expected_properties =
+            ['title', 'author', 'article_id', 'topic', 'created_at', 'votes', 'article_img_url', 'body'];
+
+          expect(Object.keys(article).length).toBe(expected_properties.length);
+
+          expected_properties.forEach(property => {
+            expect(article).toHaveProperty(property);
+          });
+        });
+    });
+
+    test('GET 404: Responds with status 404 when there is no article with specified id.', () => {
+      return request(app)
+        .get('/api/articles/999')
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("No such article");
+        });
+    });
+
+    test('GET 400: Responds with status 400 when specified id is invalid.', () => {
+      return request(app)
+        .get('/api/articles/not-a-number')
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad request");
+        });
+    });
+  });
+
+  describe('/api/articles', () => {
+    test('GET 200: Responds with an object containing a list of articles.', () => {
+      return request(app)
+      .get('/api/articles')
       .expect(200)
-      .then(({ body: { article } }) => {
-        expect(article.article_id).toBe(1);
-
-        expect(Object.hasOwn(article, 'author'));
-        expect(Object.hasOwn(article, 'title'));
-        expect(Object.hasOwn(article, 'body'));
-        expect(Object.hasOwn(article, 'topic'));
-        expect(Object.hasOwn(article, 'created_at'));
-        expect(Object.hasOwn(article, 'votes'));
-        expect(Object.hasOwn(article, 'article_img_url'));
-
-        expect(typeof article.author).toBe('string');
-        expect(typeof article.title).toBe('string');
-        expect(typeof article.body).toBe('string');
-        expect(typeof article.topic).toBe('string');
-        expect(typeof article.created_at).toBe('string');
-        expect(typeof article.votes).toBe('number');
-        expect(typeof article.article_img_url).toBe('string');
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy('created_at', { descending: true });
+
+        // Expect the article objects to have the expected set of properties and no others.
+
+        const expected_properties =
+          ['title', 'article_id', 'author', 'topic', 'created_at', 'votes', 'article_img_url', 'comment_count'];
+
+        const comment_counts = [ -1, 11, 0, 2, 0, 2, 1, 0, 0, 2, 0, 0, 0, 0 ];
+
+        articles.forEach(article => {
+          expect(article.comment_count).toBe(comment_counts[article.article_id]);
+
+          expect(Object.keys(article).length).toBe(expected_properties.length);
+
+          expected_properties.forEach(property => {
```

Comment:
Nested loops are quite inefficient in terms of time complexity, which is not really an issue here as it's only a test, but just for best practice I recommend avoiding them, which could have been done here with something like a flexible Jest matcher such as .toMatchObject or .objectContaining to see if the article has all the necessary properties.